src/cluster.py

```diff
@@ -168,11 +165,21 @@ def _create_directory(self, path: str, mode: int) -> None:
         os.chmod(path, mode)
         self._change_owner(path)
 
-    def _get_postgresql_version(self) -> str:
+    def _create_user_home_directory(self) -> None:
+        """Creates the user home directory for the snap_daemon user.
+
+        This is needed due to https://bugs.launchpad.net/snapd/+bug/2011581.
+        """
+        subprocess.run("mkdir -p /home/snap_daemon".split())
+        subprocess.run("chown snap_daemon:snap_daemon /home/snap_daemon".split())
```

Comment:
feel free to ignore, it is more jfyi how to have one command instead of two
```suggestion
        subprocess.run("install -d -u snap_daemon -g snap_daemon -m 0750 /home/snap_daemon".split())
```